File: service/src/main/java/com/iberianmotorsports/service/model/Bop.java
package com.iberianmotorsports.service.model;

import com.iberianmotorsports.service.model.composeKey.BopPrimaryKey;
import jakarta.persistence.Column;
import jakarta.persistence.EmbeddedId;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.Data;

import java.io.Serializable;

@Entity
@Data
@Table(name = "BOP")
public class Bop implements Serializable {

    @EmbeddedId
    private BopPrimaryKey bopPrimaryKey;

    @Column(name = "ballastKg", columnDefinition = "integer default 0")
    private Integer ballastKg;

    @Column(name = "restrictor", columnDefinition = "integer default 0")
    private Integer restrictor;

}

File: frontend/src/app/model/Bop.ts
import {Car} from "./Car";

export class Bop {
    raceId: number;
    car: Car;
    ballastKg: number;
    restrictor: number;

    constructor() {
        this.raceId = null;
        this.car = new Car();
        this.ballastKg = 0;
        this.restrictor = 0;
    }
}
